invenio_pidstore/providers/datacite.py

```diff
@@ -31,20 +35,66 @@ class DataCiteProvider(BaseProvider):
     default_status = PIDStatus.NEW
     """Default status for newly created PIDs by this provider."""
 
+    doi_prefix_regexp = re.compile(
+        r"10\.\d+(\.\d+)*$"
+    )
+
     @classmethod
-    def create(cls, pid_value, **kwargs):
+    def valid_doi_prefix(cls, prefix):
+        """Matches if prefix is a DOI prefix.
+
+        NOTE: This was done here to prevent relying on idutils.
+        """
+        return cls.doi_prefix_regexp.match(prefix)
+
+    @classmethod
+    def generate_doi(cls, options=None):
+        """Generate DOI with random suffix."""
+        passed_options = options or {}
+        # WHY: A new dict needs to be created to prevent side-effects
+        options = dict(current_app.config.get(
+            'PIDSTORE_DATACITE_DOI_OPTIONS', {}
+        ))
+        options.update(passed_options)
+        prefix = options.get(
+            'prefix',
+            current_app.config.get('PIDSTORE_DATACITE_DOI_PREFIX')
+        )
+        suffix_length = options.get('suffix_length')
```

Comment:
Given the signature of the method `base32.generate`, isn't is better to have "compatible" default values instead of None?
```
suffix_length = options.get('suffix_length', 8)
split_every = options.get('split_every', 0)
checksum = options.get('checksum', False)
```

Reply:
I was thinking about that too. The configuration in `config.py` provides a default was my intial thought, but we could provide additional default here as a last line of defense in case `PIDSTORE_DATACITE_DOI_OPTIONS` is misconfigured by hoster or is an empty dict for instance. Will do.